### PR TITLE
fix($core): normalize theme path with Layout.vue correctly (close #2178)

### LINKED ISSUE #2178: [Bug report] personal theme set 'layouts/Layout.vue' to main field can not load index.styl
- [x] I confirm that this is an issue rather than a question.

#### Steps to reproduce

you can see the [package.json](https://github.com/yiliang114/vuepress-theme-resume/blob/vuepress%401.x/example/.vuepress/theme/package.json) for my theme, if i just `yarn start` in local , maybe the package.json is not necessary.  The[`theme/index.js` ](https://github.com/yiliang114/vuepress-theme-resume/blob/vuepress%401.x/example/.vuepress/theme/index.js) in my theme is almost empty. And only do that as shown above, my theme can only to load index.styl automatically.

if i remove `theme/index.js` and  set the "main" field to layouts/Layout.vue in package.json from the [reference](https://vuepress.vuejs.org/theme/writing-a-theme.html#directory-structure) , after i run `npm publish ` and install the latest theme, the style can not be load in the page.

#### What is expected?
set the "main" field to layouts/Layout.vue in package.json from the [reference](https://vuepress.vuejs.org/theme/writing-a-theme.html#directory-structure) and publish my theme, people who installed the latest theme can load style correctly

#### What is actually happening?
theme may lose some style , due to the index.styl could not load automatically

#### Other relevant information

- Output of `npx vuepress info` in my VuePress project:

Environment Info:

  System:
    OS: macOS Mojave 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
  Binaries:
    Node: 10.15.2 - /usr/local/bin/node
    Yarn: 1.22.0 - ~/.yarn/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 80.0.3987.100
    Firefox: 71.0
    Safari: 13.0.4
  npmPackages:
    @vuepress/core:  1.3.0 
    @vuepress/theme-default:  1.3.0 
    vuepress: ^1.2.0 => 1.3.0 
  npmGlobalPackages:
    vuepress: 1.2.0

### PR DESCRIPTION
**Pull request by meteorlxy**
**Summary**

Fix #2178 

Caused by #1564

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

**Other information:**

### CODE REVIEW

File: packages/@vuepress/core/lib/node/loadTheme.js [hunk 1]
```diff
@@ -60,8 +60,15 @@ module.exports = function loadTheme (ctx) {
 }
 
 function normalizeThemePath (resolved) {
-  const { entry, fromDep } = resolved
+  const { entry, name, fromDep } = resolved
   if (fromDep) {
+    const packageRoot = require.resolve(`${name}/package.json`)
+    const { main } = require(packageRoot)
+    if (main.endsWith('.vue')) {
```
Comment (kefranabg) [inline packages/@vuepress/core/lib/node/loadTheme.js]: Could this break if `main` field is not defined in `package.json` ?

Comment (meteorlxy) [reply packages/@vuepress/core/lib/node/loadTheme.js]: Uh you are right. I just reverted this code snippet from #1564

Comment (meteorlxy) [reply packages/@vuepress/core/lib/node/loadTheme.js]: Tweaked

**Comment by newsbielt703** [comment]
Testing it with the [reproduction link](https://github.com/yiliang114/vuepress-reproduction-demo) from #2178, I somehow got the error `Error: Cannot find module 'vuepress-theme-md-resume/package.json ...`

**Pull request closed without merging**